assets/js/filter-component.js

```diff
@@ -0,0 +1,118 @@
+;(function setupFilterComponent() {
+  'use strict'
+
+  function resizeFilterSections() {
+    var filterButtons = document.getElementsByClassName('bapv-filter-button')
```

Comment:
I wonder, could these classname be passed in somehow? I realise there's not much going on with it elsewhere but it'd be nice to make it easier to setup, i.e. just include a file and set some variables instead of trying to find the classnames in the code?

Reply:
Had a look at class names suggestion. Not sure it would ultimately be simpler though. I'm seeing the JS (at assets/js/filter-component.js) and associated Nunjucks template macro (at `server/views/components/filter/template.njk`) as collectively the 'BaPV filter component', so from the main application code you only have to do:
```
{{ bapvFilter({...}) }}
``` 
to use it. I guess it's just the same as the coupling between CSS/JS/HTML in the MoJ / GDS components - but you don't normally see the details as the files are all in `node_modules` 🤷‍♂️!

Have taken on board the point about use of `var` and similar so have freshened up the JS/CSS 🙂.